eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Ratify.hs

```diff
@@ -203,6 +208,23 @@ delayingAction TreasuryWithdrawals {} = False
 delayingAction ParameterChange {} = False
 delayingAction InfoAction {} = False
 
+actionPriority :: GovAction era -> Int
+actionPriority NoConfidence {} = 0
+actionPriority NewCommittee {} = 1
+actionPriority NewConstitution {} = 2
+actionPriority HardForkInitiation {} = 3
+actionPriority ParameterChange {} = 4
+actionPriority TreasuryWithdrawals {} = 5
+actionPriority InfoAction {} = 6
+
+reorderActions :: StrictSeq (GovActionState era) -> StrictSeq (GovActionState era)
+reorderActions acts =
+  Seq.fromList . concat . IntMap.elems $
+    foldr'
+      (\x -> IntMap.insertWith (<>) (actionPriority $ gasAction x) [x])
+      IntMap.empty
+      acts
```

Comment:
Is this more efficient than sorting, for example:
`Seq.fromList $ sortOn (actionPriority . gasAction)  (toList acts)`     ?

Reply:
Yes. `sortOn` and `sort` are pretty slow. Using `Map` and `IntMap` and `Set` for sorting will always be much faster.

Note also an important fact that it would have to be `nubOn . sortOn` in order to match `IntSet.elems . IntMap.fromList`

Reply:
Also I don't think `sort` is guaranteed to preserve the order of equal elements.

Reply:
@Soupstraw `Data.List.sort` implements a stable sorting algorithm, so equal elements are guaranteed to preserve relative order.

Reply:
That being said, because of potential inefficiency: ` IntMap.insertWith (<>) (actionPriority $ gasAction x) [x]` I am not really sure that in this whole approach is really worth it. Using `sortOn` will be much simpler.

In the matter of fact this whole PR can be made much more efficient, but we don't have to deal with that right now.

Reply:
So, let's switch it to `sortOn`, so it clearer what is going on.